pkg/operator/ceph/nfs/spec.go

```diff
@@ -141,6 +141,12 @@ func (r *ReconcileCephNFS) makeDeployment(nfs *cephv1.CephNFS, cfg daemonConfig)
 		ObjectMeta: metav1.ObjectMeta{
 			Name:   resourceName,
 			Labels: getLabels(nfs, cfg.ID, true),
+			Annotations: map[string]string{
```

Comment:
You're seeing an update to the annotations cause a pod restart? I was thinking the annotations didn't affect the pod restart, but that must only be the case when the annotations are on the deployment instead of the podTemplateSpec.

Reply:
I thought I had a similar comment in my initial review, somehow I cannot find it anymore. Why not use the deployment annotation instead of the pod's one?

Reply:
Because a changed annotation on the deployment won't cause the pod to restart, but a changed annotation on the pod should.

Reply:
Confirmed that the annotation causes a pod restart as desired.

Reply:
One more question, the configmap is updated during the reconcile right? So regardless of the annotation being on pod or the deployment, the pod is going to be restarted right? If this is true I don't understand why we don't place the annotation on the deployment instead.
Quickly glancing at the code, I doubt the CephNFS config should be changed or supports any changes despite `GaneshaServerSpec`. The only CR spec that impacts the configmap is `n.Spec.RADOS`, especially `n.Spec.RADOS.Namespace`, is it safe to change it? Since recently changing `n.Spec.RADOS.Pool` does not do anything since it always defaults to `.nfs`.
If all the above is true and changing `n.Spec.RADOS.Namespace` is risky then I'm wondering why we would need to restart the pod if the config changes if there is a misconfiguration in the CR perhaps? I just don't see where it could be.
Thanks.